### PR TITLE
Items management #141 puzlle

### PR DESCRIPTION
**Pull request by ammaratef45**
Solved Puzzle #141 by:
1- Creating views file.
2- Creating tests.
3- Creating templates.
4- Leaving a puzzle to continue the implementation.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#253](https://github.com/timelesslounge/timelessis/issues/253) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#253](https://github.com/timelesslounge/timelessis/issues/253) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/25a6143d-948e-4e89-afa4-0672fc4b106e); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: timeless/items/views.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+@BP.route("/edit", methods=("GET", "POST"))
+def edit():
+    """ Edit an item by id """
+    return render_template("items/edit.html")
+
+@BP.route("/delete", methods=["POST"])
+def delete():
+    """ Delete an item by id """
+    return redirect(url_for("items.list_items"))
```
Comment (amihaiemil) [inline timeless/items/views.py]: @ammaratef45 These endpoints should require authentication, right? If you have time, search other views (Companies, I think) which have the authentication annotation. If not, just leave a puzzle :)

File: timeless/templates/items/list.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+{% block header %}
+    <h1>{% block title %}Items{% endblock %}</h1>
+{% endblock %}
+
+{% block content %}
+    {% for item in items %}
+        <article class="item">
+        <header>
+            <div>
+            <h1>{{ item['id'] }}</h1>
```
Comment (amihaiemil) [inline timeless/templates/items/list.html]: @ammaratef45 Do we only display the item's ID?

File: timeless/templates/items/list.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            </div>
+            <a class="action" href="{{ url_for('items.edit', id=item['id']) }}">Edit</a>
+            <a class="action" href="{{ url_for('items.delete', id=item['id']) }}">Delete</a>
+        </header>
+        </article>
+        {% if not loop.last %}
+        <hr>
+        {% endif %}
+    {% endfor %}
+{% endblock %}
```
Comment (amihaiemil) [inline timeless/templates/items/list.html]: @ammaratef45 new line at the end of file please

File: timeless/templates/items/list.html (hunk 2 above)
Comment (ammaratef45) [reply timeless/templates/items/list.html]: The templates should be continued as in the puzzle I left.

**Comment by ammaratef45** [comment]
@amihaiemil Thanks, please take a look.

**Comment by ammaratef45** [comment]
@amihaiemil The test file was deleted in the PR #258 
That makes a conflict with my PR

**Comment by amihaiemil** [comment]
@ammaratef45 
> That makes a conflict with my PR

Yes, you have to fix it locally, I think. It may be too complex for the Github Browser tool

**Comment by ammaratef45** [comment]
@amihaiemil Yes, I don't even have the needed access to solve it here
The question is, should I just re-add the file? I don't understand why it was deleted

**Comment by amihaiemil** [comment]
@ammaratef45 it was moved or renamed, not actually deleted. Download Github Desktop, there you will see exactly what the conflict is and what to do. You probably simply have to delete your local file and commit it (the new, renamed/moved version will come with the merge).

**Comment by ammaratef45** [comment]
@amihaiemil in commit ab7e0c14849d6c5fe266986e1d045210bcd0e52f the file was deleted and it's content was moved back to test_models file.
I have added it again and deletd the old content.
I also merged the master branch.
But I get `ModuleNotFoundError: No module named '_bz2'` when trying to run pytest.

**Comment by amihaiemil** [comment]
@ammaratef45 try to see if there wasn't any residual file created, by any chance? And search in the project to see what ``_bz2`` actually is -- maybe it's a new dependency that you have to install in your environment

**Comment by ammaratef45** [comment]
@amihaiemil [this ](https://stackoverflow.com/a/12806325/5565834)solved my problem, had to rebuild python.
Can you take a look now?

**Comment by amihaiemil** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 30 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#253](https://github.com/timelesslounge/timelessis/issues/253) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one major issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by amihaiemil** [comment]
@ypshenychka confirmed

**Comment by ypshenychka** [comment]
@amihaiemil thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
> @0crat quality acceptable ([here](https://github.com/timelesslounge/timelessis/issues/253#issuecomment-464301508))

@ypshenychka There is no quality review for [#253](https://github.com/timelesslounge/timelessis/issues/253), no performer

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**